R/detect_mistakes.R

```diff
@@ -203,9 +204,30 @@ detect_mistakes <- function(user,
       )
     }
 
+    
+    if (length(well_matched > 0)){
+      matched_user_names <- sort(rlang::names2(well_matched))
+      
+    if ( !allow_partial_matching ){
+      
+      matches <- which(sapply(lapply(remaining_solution_names,startsWith, matched_user_names),sum)==1)
```

Comment:
When possible, please use `vapply` over `sapply`. 

`sapply` is not consistent in its return shape.

Reply:
I am confused on this logic. Can you explain what you are trying to accomplish?

Reply:
I change to use vapply()